src/carousel/index.ts

```diff
@@ -0,0 +1,222 @@
+    this.set({ transitioning: false });
+  }
+
+  trackStyle = () => {
+    if (this.state.transitioning && this.state.settings.speed) {
+      const { speed } = this.state.settings;
+      const transition = speed + 'ms ' + 'ease';
+      const transitionStyles = {
+        '-webkit-transition': transition,
+        transition,
+        '-ms-transition': transition
+      };
+      return { ...this.getStaticTrackStyle(), ...transitionStyles };
+    } else {
+      return this.getStaticTrackStyle();
+    }
+  }
+
+  getStaticTrackStyle = () => {
+    const slideWidth = this.getSlideWidth();
+    const slideCount = this.props.items.length;
+    const slidesToShow = this.state.settings.slidesToShow;
+    const trackWidth = (slideCount + 4 * slidesToShow - 2) * slideWidth;
+    const pos = Carousel.calculatePosition(this.state.currentSlide, slideWidth, slidesToShow);
+    const transform = `translateX(-${pos}px)`;
+    const transformStyles = {
+      transform,
+      '-webkit-transform': transform,
+      '-ms-transform': transform
+    };
+
+    return { width: `${trackWidth}px`, ...transformStyles };
+  }
+
+  getSlideWidth = () => {
+    const visibleWidth = this.refs.wrapper.offsetWidth;
+
+    if (visibleWidth) {
+      return visibleWidth / this.state.settings.slidesToShow;
+    }
+  }
+
+  onTouchStart = (e: TouchEvent & Carousel.Event) => {
+    e.stopPropagation();
+
+    this.state.touchObject = {
+      startX: e.touches[0].pageX,
+      startY: e.touches[0].pageY
+    };
+
+    e.target.addEventListener('touchend', this.onTouchEnd);
+  }
+
+  onTouchEnd = (e: TouchEvent & Carousel.Event) => {
+    const curX = e.changedTouches[0].pageX;
+    const curY = e.changedTouches[0].pageY;
+    // swipe distance needs to be more than 20
+    if (Math.abs(curX - this.state.touchObject.startX) > 20) {
+      Carousel.isSwipeToNext({ ...this.state.touchObject }, curX, curY) ? this.moveNext() : this.movePrevious();
+    }
+
+    e.target.removeEventListener('touchstart', this.onTouchStart);
+    e.target.removeEventListener('touchend', this.onTouchEnd);
+  }
+
+  static calculatePosition = (currentSlide: number, moveDistance: number, slidesToShow: number): number =>
+    (currentSlide + slidesToShow * 2 - 1) * moveDistance
+
+  static isSwipeToNext = (touchObj: Carousel.TouchObject, curX: number, curY: number): boolean => {
+    const xDist = touchObj.startX - curX;
+    const yDist = touchObj.startY - curY;
+
+    const r = Math.atan2(yDist, xDist);
+
+    let swipeAngle = Math.round(r * 180 / Math.PI);
+
+    if (swipeAngle < 0) {
+      swipeAngle = 360 - Math.abs(swipeAngle);
+    }
+    // swipeAngle between 45 and 135, between 225 and 315 is ignored;
+    if (
+      (swipeAngle <= 45 && swipeAngle >= 0) ||
+      (swipeAngle <= 360 && swipeAngle >= 315)
+    ) {
+      return true;
+    } else if (swipeAngle >= 135 && swipeAngle <= 225) {
+      return false;
+    }
```

Comment:
What is returned when swipeAngle is between 45 and 135? Similarly when it's between 225 and 315?

If returning undefined for false is acceptable then we don't need the last 'else if' block either.